CIPs/cip-137.md

```diff
@@ -18,187 +18,223 @@ An API for reading and writing events on Ceramic event streams.
 
 ## Abstract
 <!--A short (~200 word) description of the technical issue being addressed.-->
-This CIP introduces a JSON-RPC interface that can be used to subscribe to a number of streams inside of an interest range, and write new events to these streams. The API exposes two main ways of retrieving events. A simple method that uses the nodes local ordering of events to retrieve events by simply keeping track of a single vector client side. More advanced users can use a simplified version of the [Recon](https://cips.ceramic.network/CIPs/cip-124) protocol to retrieve events in their logical order.
+This CIP introduces a REST API that can be used to subscribe to a number of streams inside of an interest range, and write new events to these streams. The API exposes two main ways of retrieving events. A simple method that uses the nodes local ordering of events to retrieve events by simply keeping track of a single vector client side. More advanced users can use a simplified version of the [Recon](https://cips.ceramic.network/CIPs/cip-124) protocol to retrieve events in their logical order.
 
 
 ## Motivation
 <!--Motivation is critical for CIPs that want to change the Ceramic protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the CIP solves. CIP submissions without sufficient motivation may be rejected outright.-->
-In order to enable databases other than ComposeDB to be built on top of Ceramic, an API to read and write directly on event streams is needed. This also enables developers to get direct access to the event streams underlaying ComposeDB.
+In order to enable services and databases other than ComposeDB to be built on top of Ceramic, an API to read and write directly on event streams is needed. This also enables developers to get direct access to the event streams underlaying ComposeDB.
 
 
 ## Specification
 <!--The technical specification should describe the syntax and semantics of any new feature.-->
 
-This is a work in progress specification that introduces a JSON-RPC interface for interacting directly with event streams on a node.
+This specification introduces a REST API for interacting directly with event streams on a node.
 
 
 
-### `ceramic_subscribe`
+### GET `/ceramic/subscribe/{streamid}`
```

Comment:
This should be a POST, even if we use query parameters for specifying the subscription.

Reply:
A get to this endpoint would return information about the subscription.

Reply:
Not sure I follow. Why should it be a POST?

Reply:
We're changing the state of the system, so should be one of POST/PUT/PATCH, with POST the best verb to use.

Reply:
Depends on which system you are referring to. The node is simply just giving you information directly from the network. If it has this info locally already it will just give it to you, otherwise sync it from the network.